cmd/aws-s3-proxy/pkg/serve.go

```diff
@@ -28,112 +29,175 @@ var serveCmd = &cobra.Command{
+	viperBindFlag("corsmaxage", serveCmd.Flags().Lookup("cors-max-age"))
+
+	serveCmd.Flags().IntP("max-idle-connections", "", 150, "max idle connections")
+	viperBindFlag("maxidleconnections", serveCmd.Flags().Lookup("max-idle-connections"))
+
+	serveCmd.Flags().StringP("basic-auth-user", "", "", "username for basic auth")
+	viperBindFlag("basicauthuser", serveCmd.Flags().Lookup("basic-auth-user"))
+
+	serveCmd.Flags().StringP("cors-allow-headers", "", "", "CORS: Comma-delimited list of the supported request headers")
+	viperBindFlag("corsallowheaders", serveCmd.Flags().Lookup("cors-allow-headers"))
+
+	serveCmd.Flags().StringP("cors-allow-methods", "", "", "CORS: comma-delimited list of the allowed - https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html")
+	viperBindFlag("corsallowmethods", serveCmd.Flags().Lookup("cors-allow-methods"))
+
+	serveCmd.Flags().StringP("cors-allow-origin", "", "", "CORS: a URI that may access the resource")
+	viperBindFlag("corsalloworigin", serveCmd.Flags().Lookup("cors-allow-origin"))
+
+	serveCmd.Flags().StringP("healthcheck-path", "", "", "path for healthcheck")
+	viperBindFlag("healthcheckpath", serveCmd.Flags().Lookup("healthcheck-path"))
+
+	serveCmd.Flags().StringP("listen-address", "", "::1", "host address to listen on")
+	viperBindFlag("listenaddress", serveCmd.Flags().Lookup("listen-address"))
+
+	serveCmd.Flags().StringP("listen-port", "", "21080", "port to listen on")
+	viperBindFlag("listenport", serveCmd.Flags().Lookup("listen-port"))
+
+	serveCmd.Flags().StringP("http-cache-control", "", "", "overrides S3's HTTP `Cache-Control` header")
+	viperBindFlag("httpcachecontrol", serveCmd.Flags().Lookup("http-cache-control"))
+
+	serveCmd.Flags().StringP("http-expires", "", "", "overrides S3's HTTP `Expires` header")
+	viperBindFlag("httpexpires", serveCmd.Flags().Lookup("http-expires"))
+
+	serveCmd.Flags().StringP("index-document", "", "index.html", "the index document for static website")
+	viperBindFlag("indexdocument", serveCmd.Flags().Lookup("index-document"))
+
+	serveCmd.Flags().StringP("upstream-bucket", "", "", "upstream s3 bucket")
+	viperBindFlag("s3bucket", serveCmd.Flags().Lookup("upstream-bucket"))
+
+	serveCmd.Flags().StringP("upstream-key-prefix", "", "", "upstream s3 path/key prefix")
+	viperBindFlag("s3prefix", serveCmd.Flags().Lookup("upstream-key-prefix"))
+
+	serveCmd.Flags().StringP("ssl-cert-path", "", "", "path to ssl cert")
+	viperBindFlag("sslcert", serveCmd.Flags().Lookup("ssl-cert-path"))
+
+	serveCmd.Flags().StringP("ssl-key-path", "", "", "path to ssl key")
+	viperBindFlag("sslkey", serveCmd.Flags().Lookup("ssl-key-path"))
+
+	serveCmd.Flags().StringP("strip-path", "", "", "strip path prefix")
+	viperBindFlag("strippath", serveCmd.Flags().Lookup("strip-path"))
+
+	if err := serveCmd.MarkFlagRequired("upstream-bucket"); err != nil {
+		logger.Fatal(err)
+	}
+
+	if len(os.Getenv("S3_PROXY_BASIC_AUTH_PASS")) != 0 {
```

Comment:
shouldn't this be set as part of the `AutomaticEnv`?

Reply:
from what I was testing, since it isn't in the flags, no. and I don't want to make a secret a flag.

Reply:
I thought the `AutomaticEnv` would pull in any variable with the prefix, in this case `S3_PROXY` and you could just do: 

```
viper.Get("basic-auth-pass")
```